### PR TITLE
Fix IllegalArgumentException: Cannot copy null stack in ItemStackMixin_NeoForge

### PR DESCRIPTION
**Pull request by LabsZero**
Fixes IllegalArgumentException: Cannot copy null stack reported in server logs.

### CODE REVIEW

File: arclight-neoforge/src/main/java/io/izzel/arclight/neoforge/mixin/core/world/item/ItemStackMixin_NeoForge.java [hunk 1]
```diff
@@ -64,7 +69,7 @@ public abstract class ItemStackMixin_NeoForge implements ItemStackBridge, IItemS
     }
 
     @Deprecated
-    public void setItem(Item item) {
+    public void setItem(@Nullable Item item) {
```
Comment (qyl27) [inline arclight-neoforge/src/main/java/io/izzel/arclight/neoforge/mixin/core/world/item/ItemStackMixin_NeoForge.java:70]: As far as I know, there shouldn't be an `ItemStack` with a null item, it should be `Items.AIR` instead.

File: arclight-neoforge/src/main/java/io/izzel/arclight/neoforge/mixin/core/world/item/ItemStackMixin_NeoForge.java [hunk 2]
```diff
@@ -38,8 +38,13 @@ public abstract class ItemStackMixin_NeoForge implements ItemStackBridge, IItemS
     // @formatter:on
 
     @Decorate(method = "hurtAndBreak(ILnet/minecraft/server/level/ServerLevel;Lnet/minecraft/world/entity/LivingEntity;Ljava/util/function/Consumer;)V",
-        at = @At(value = "INVOKE", target = "Lnet/minecraft/world/item/enchantment/EnchantmentHelper;processDurabilityChange(Lnet/minecraft/server/level/ServerLevel;Lnet/minecraft/world/item/ItemStack;I)I"))
+            at = @At(value = "INVOKE", target = "Lnet/minecraft/world/item/enchantment/EnchantmentHelper;processDurabilityChange(Lnet/minecraft/server/level/ServerLevel;Lnet/minecraft/world/item/ItemStack;I)I"))
     private int arclight$itemDamage(ServerLevel serverLevel, ItemStack itemStack, int i, @Local(ordinal = 0) LivingEntity damager) throws Throwable {
+        // Added null check for itemStack
```
Comment (qyl27) [inline arclight-neoforge/src/main/java/io/izzel/arclight/neoforge/mixin/core/world/item/ItemStackMixin_NeoForge.java]: Would you mind figuring out which method passed a null as `ItemStack` into this method?  
In Minecraft modding, these should all be replaced by `ItemStack,EMPTY`. If it happened in NeoForge, perhaps it was their bug.

**Review by qyl27 — CHANGES_REQUESTED** [review]

File: arclight-neoforge/src/main/java/io/izzel/arclight/neoforge/mixin/core/world/item/ItemStackMixin_NeoForge.java (hunk 2 above)
Comment (LabsZero) [reply arclight-neoforge/src/main/java/io/izzel/arclight/neoforge/mixin/core/world/item/ItemStackMixin_NeoForge.java]: Sure i will look into it asap

**Comment by LabsZero** [comment]
ignore my last pull request still looking into the matter.

**Review by LabsZero — COMMENTED** [review]
Hopefully this fixes the null issue. If it needs more than happy to work on it further.

**Review by qyl27 — APPROVED** [review]

**Comment by qyl27** [comment]
Although I am still confused as to why the `this` of `ItemStack` will be null, thanks for your PR, I'll merge it.

**Comment by qyl27** [comment]
I think the root problem is in https://github.com/IzzelAliz/Arclight/blob/ab1e5d2eba758556f969e88d5f5f6967ee18f5f4/arclight-common/src/main/java/io/izzel/arclight/common/mod/util/ArclightSpecialRecipe.java#L21
And I have solved it.

**Pull request merged**